src/main/java/messenger/MessengerDatabase.java

```diff
@@ -0,0 +1,43 @@
+package messenger;
+
+import notification.Notification;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MessengerDatabase {
+
+    private HashMap<String, String> idToNameMap;
```

Comment:
make type the less specific - `Map<String, String>`, instead of `HashMap`

Reply:
Why should I?

Reply:
This is what polymorphism and inheritance is all about. You use method from not-specific interface (`Map`) and choose Implementation (`HashMap`) once. Then switching implementations are easy.

Suppose you want to change HashMap to TreeMap in the future. When using it like this you will get a ton of type mismatches. For this small application it's less of a problem since these occurrences are limited. When this project grows bigger you will find yourself using this a lot more. Additionally just `Map` and `List` and `Set` are always the types of return parameters of all of the libraries. By keeping you variable as `HashMap` you need to perform casting every now and then. Just like few lines below.